### DISCUSSION TITLE
Disabling Talk and/or Collabora Servers through AIO causes the respective Apps to be uninstalled automatically from Nextcloud

### DISCUSSION THREAD

**Discussion by lollo0296** [nextcloud/all-in-one#2568] [Questions] [answered]
### Steps to reproduce
1. Install the "Talk" or "Nextcloud Office" App through the Nextcloud GUI
2. Make sure that the respective backend Service it is not enable through AIO
3. Wait until AIO does it's magic (containers restart, update or backup)

### Expected behavior
Disabling the backend service (or not ever having it enabled in the first place) should not uninstall the respective app.

### Actual behavior
Once everything's back up and running after step 3, both Apps are gone.

We configured a "talk high performance backend" together with "Collabora server" on a different host from AIO.

For Talk, we never used it within AIO before, so I just downloaded the app and configured all the URLs and secrets in the administration settings.

For Collabora, we had the server enabled in AIO, so I just changed the server URL in the administration settings. I later unchecked Collabora in the AIO-GUI.

Every day both apps are gone. The good side of things ist that all the administrative settings and the Talk data is not being lost despite the apps being uninstalled.
Now I have to manually re-install both apps every day LOL (not so funny actually 😅).

### Host OS
Debian 11, latest versions of everything 

#### Nextcloud AIO version
v5.1.0

#### Current channel
Latest

**Comment by lollo0296** [comment]
@szaimen Heeeelp please 😭🙏🏼😵‍💫🤣🤣

**Comment by szaimen** [comment]
Hi, as for talk you could leave the container enabled. It should not overwrite your custom settings. 

As for collabora is this a bit more complicated.

Btw what are the reasons you are running the services externally?

**Reply by lollo0296** [reply]
Talk: leaving the container running without being used sounds like a waste of resources to me. In addition, this might also cause confusion in the future, specially for others admins who find the service enabled and the container running.

The reason why we decided to deploy a cloud based high performance backend within a dedicated VPS in the first place can be found in [this video by @apfelcast](https://youtu.be/lMaR8zStTDU).

(For non-german speakers, the guy in the video states that `you need an high performance backend on a different host if you want to reliable use talk with more than 4 participants at the same time`).

Is this statement incorrect?

In the video, he showcases a very nice script that easily allows to install and set up a talk high performance backend and (if desired) a Collabora server on a Debian server. The scripts also take cares of backups and unattended updates and can be found [here](https://github.com/sunweaver/nextcloud-high-performance-backend-setup).

So I went ahead and deployed talk HPB and Collabora server with the mentioned script on a fresh Debian VPS, which is ultimately the reason why I encountered the problem with the Talk and Nextcloud Office App within our Nextcloud AIO-installation. 🤷🏻‍♂️

**Comment by szaimen** [comment]
> (For non-german speakers, the guy in the video states that `you need an high performance backend on a different host if you want to reliable use talk with more than 4 participants at the same time`).
> 
> Is this statement incorrect?

This is indeed incorrect. If your server is powerful enough, why should it not be able to run the talk hpb on the same server? This makes no sense. BTW, see https://github.com/nextcloud/all-in-one/discussions/1335 if you have not seen it yet.

**Reply by lollo0296** [reply]
Ok, thank you for the clarification.

Back to the main point of the conversation, this means that **a Nextcloud installation deployed through AIO is not compatible at all with external HPB and/or external Collabora server.** That's a pity, isn't it?
I mean, there might be a lot of reason why someone wanted to actually run those services on a different machine (performance/available resources being just one of them). **_Wouldn't be better to at least support this practice within AIO?_**

**_My suggestion:_** AIO could, for example, still install and configure the app if the related service is enabled on the AIO-UI, but just leave everything as it is if the service is unchecked (not enabled).
A step further could be to show a popup/modal when unchecking the talk and/or Collabora service in AIO, informing the administrator that the related app must be uninstalled manually if no longer needed.

So last but not least, I will try to use the integrated Talk HPB of AIO instead of the external one. I have one question for you:

- Our Nextcloud installation runs behind a reverse proxy, which takes care of encryption. As I understand, for the Talk HPB the Port 3478 TCP+UDP should be directly port-forwarded toward the Talk container and not go through a proxy. No need for TLS certificates here?

Thank you so much

**Reply by szaimen** [reply]
> * As I understand, for the Talk HPB the Port 3478 TCP+UDP should be directly port-forwarded toward the Talk container and not go through a proxy. No need for TLS certificates here?

Yes, this is correct.

**Reply by lollo0296** [reply]
Got it, thank you!

> Back to the main point of the conversation, this means that a Nextcloud installation deployed through AIO is not compatible at all with external HPB and/or external Collabora server. That's a pity, isn't it?
> I mean, there might be a lot of reason why someone wanted to actually run those services on a different machine (performance/available resources being just one of them). Wouldn't be better to at least support this practice within AIO?
> 
> My suggestion: AIO could, for example, still install and configure the app if the related service is enabled on the AIO-UI, but just leave everything as it is if the service is unchecked (not enabled).
> A step further could be to show a popup/modal when unchecking the talk and/or Collabora service in AIO, informing the administrator that the related app must be uninstalled manually if no longer needed.

What do you think about this topic? Is it worth considering or not?

**Reply by lollo0296** [reply]
@szaimen What do you think about this topic? Is it worth considering or not?

> [...]a Nextcloud installation deployed through AIO is not compatible at all with external HPB and/or external Collabora server. That's a pity, isn't it?
> I mean, there might be a lot of reason why someone wanted to actually run those services on a different machine (performance/available resources being just one of them). Wouldn't be better to at least support this practice within AIO?
> 
> My suggestion: AIO could, for example, still install and configure the app if the related service is enabled on the AIO-UI, but just leave everything as it is if the service is unchecked (not enabled).
> A step further could be to show a popup/modal when unchecking the talk and/or Collabora service in AIO, informing the administrator that the related app must be uninstalled manually if no longer needed.
>

**Reply by szaimen** [reply]
Currently this is not planned.

**Comment by szaimen** [comment, marked as answer]
This is now possible with v7.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
